### PR TITLE
36: add Subscription.addToComposite() to BasePresenter

### PR DESCRIPTION
**Pull request by jshvarts**

### CODE REVIEW

File: app/src/main/kotlin/com/exallium/todoapp/allnotes/AllNotesPresenter.kt [hunk 1]
```diff
@@ -23,8 +23,8 @@ class AllNotesPresenter(private val view: AllNotesView,
         view.setAdapter(adapter)
         adapter.requestUpdate()
 
-        compositeSubscription.add(adapter.noteClicks().subscribe(showNoteSubscriberFn))
-        compositeSubscription.add(view.addNoteClicks().map { null }.subscribe(showNoteSubscriberFn))
+        adapter.noteClicks().subscribe(showNoteSubscriberFn).addToComposite()
+        view.addNoteClicks().map { null }.subscribe(showNoteSubscriberFn).addToComposite()
```
Comment (exallium) [inline app/src/main/kotlin/com/exallium/todoapp/allnotes/AllNotesPresenter.kt:27]: =D

File: app/src/main/kotlin/com/exallium/todoapp/mvp/BasePresenter.kt [hunk 2]
```diff
@@ -61,4 +62,6 @@ abstract class BasePresenter<out V : BaseView>(view: V) {
             }
         }
     }
-}
\ No newline at end of file
+
+    fun Subscription.addToComposite() { compositeSubscription.add(this) }
```
Comment (exallium) [inline app/src/main/kotlin/com/exallium/todoapp/mvp/BasePresenter.kt]: excellent.  Can we make this protected?

**Review by exallium — CHANGES_REQUESTED** [review]
One quick change.

File: app/src/main/kotlin/com/exallium/todoapp/mvp/BasePresenter.kt (hunk 2 above)
Comment (jshvarts) [reply app/src/main/kotlin/com/exallium/todoapp/mvp/BasePresenter.kt]: done. great idea

**Review by exallium — APPROVED** [review]

**Pull request merged**